core/autoscaling.go

```diff
@@ -63,6 +64,37 @@ const (
 	// DefaultBiddingPolicy stores the default bidding policy for
 	// the spot bid on a per-group level
 	DefaultBiddingPolicy = "normal"
+
+	// DefaultSIRRequestCompleteTagName is the name of the Tag
+	// used to mark that the Spot Instance Request has been satisfied
+	// And the new sport instance has been added to the asg for replace the
+	// pre-existing on demand instance
+	DefaultSIRRequestCompleteTagName = "autospotting-complete"
```

Comment:
Why do we need to keep track of this by tagging?

We can simply check if the SIR has an instance and if the instance is part of the current ASG. With the 5 min default Lambda run frequency the next run will start only after the SIR was expired.